cs50/2017/ap/finder/check50/__init__.py

```diff
@@ -0,0 +1,64 @@
+from check50 import *
+
+
+class Finder(Checks):
+
+    @check()
+    def exists(self):
+        """finder.c exists"""
+        self.require("finder.c")
+
+    @check("exists")
+    def compiles(self):
+        """finder.c compiles"""
+        self.spawn("clang -o finder finder.c -lcs50 -lm").exit(0)
+
+    @check("compiles")
+    def test_handles_lack_of_arguments(self):
+        """handles lack of arguments"""
+        self.spawn("./finder").exit(1)
+
+    @check("compiles")
+    def test_handles_too_many_arguments(self):
+        """handles too many arguments"""
+        self.spawn("./finder x y z").exit(1)
+
+    @check("compiles")
+    def test_cats(self):
+        """finds cats in cats.txt"""
+        self.add("foo"), self.add("bar"), self.add("this"), self.add("cats.txt"), self.add("dogs.txt"), self.add("001.txt")
```

Comment:
You can do `self.add("foo", "bar", "his", "cats.txt", "dogs.txt", "001.txt")` here, and every where else you have several `self.add`s in a row.

Reply:
Also, protip: if some of those files are required by lots of the checks, you can `add` them in e.g. `exists` and since they all depend on it, they'll get those files automatically. Saves you from having to `add` the same things over and over